predicate/solver/teleport.py

```diff
@@ -188,26 +188,48 @@ def map_policies(policy_names, policies):
     return PolicySet(mapped_policies)
 
 
-def reviews(*roles: tuple):
+def reviews(expr, *policies):
     """
     Reviews converts qualified reviews into a list ("review", "review")
     reviews((devs, expr)) -> ("review")
     """
 
-    def iff(iterator):
-        try:
-            role, expr = next(iterator)
-        except StopIteration:
-            return ast.StringTuple(())
-        else:
-            return ast.If(
-                # this is to collect all subexpressions related to review, if any
-                role.build_predicate(Review(ast.BoolLiteral(True))).expr,
-                ast.StringTuple.cons("review", iff(iterator)),
-                iff(iterator),
-            )
+    p = ast.Predicate(expr)
+    ret, model = p.solve()
+    if not ret:
+        return None
+    vals = model.eval(RequestPolicy.names.traverse())
+
+    out = []
+    for i in range(model.eval(ast.fn_string_list_len(vals)).as_long()):
```

Comment:
what does this block do? I found it somewhat unclear to follow. Comments?

Reply:
Added comments.